pkg/operation/botanist/component/dependencywatchdog/bootstrap.go

```diff
@@ -17,6 +17,9 @@ package dependencywatchdog
 import (
 	"context"
 	"fmt"
+
+	yaml "github.com/goccy/go-yaml"
```

Comment:
currently if you see go.mod you will find many libraries for yaml marshall/unmarshall. This is yet another one. The specific reason given is the marshalling of `*time.Duration` converts it into nano seconds which is quite unreadable as it is part of the configmap. Will test it and update.

Reply:
https://go.dev/play/
using gopkg.in/yaml.v3 works as expected. What was the issue that you faced?

Reply:
I didn't try with `gopkg.in/yaml.v3` . Yes it works like `github.com/goccy/go-yaml` only. I will do the replacement then for both DWD and gardener changes to use `gopkg.in/yaml.v3`
While on using `github.com/ghodss/yaml` we don't get the marshalled result but time in nanoseconds.
https://go.dev/play/p/XuXB1LTdOD0

Reply:
😵 I don't get it, let's please not introduce yet another dependency just for YAML marshalling. As already said above, there are plenty of other packages already that should provide this functionality.

Reply:
@rfranzke can we upgrade from existing gopkg.in/yaml.v2 v2.4.0 to gopkg.in/yaml.v3 ?
The later version has a lot of improvements.

Reply:
I think this should be possible, yes.

Reply:
Ok i had a call with Himanshu and the issue is that  `gopkg.in/yaml.v3` do not support `json` tags. This is semantically correct as JSON is not YAML but unfortunately we cannot change the struct definition that we borrow from autoscaling (v1/CrossVersionObjectReference) and the authors of `gopkg.in/yaml.v3` are fairly rigid - see https://github.com/go-yaml/yaml/issues/424 so they are not inclined to support treating json tags as yaml.
Other libs that are currently used in g/g have an issue that time.Duration is not serialized as is but is converted to a unreadable nano seconds value which is quite strange to me. 

@rfranzke  So we are stuck between a rock and a hard place. The only cheaper alternative is to use `yaml.v3` and then dont use `v1/CrossVersionObjectReference` and define our own struct for it with proper yaml tags. WDYT?

Reply:
Another way is we simply ditch YAML and just stick to JSON as suggested by @himanshu-kun

Reply:
What about reusing the Kubernetes API machinery like we do it for the component configs of other components (e.g. https://github.com/gardener/gardener-extension-provider-aws/tree/master/pkg/apis/config)? This way, you inherit all benefits and best-practices (versioning, encoding, etc.)

Reply:
Thanks, I'll try using it.